### ISSUE TITLE
Missed/shifted boxes problem

### ISSUE THREAD

**Issue by eslambakr** [MukhlasAdib/CARLA-2DBBox#5]
Thanks for your amazing work.

I have noticed two main problems in your module:
1) The first one is the shifted boxes as shown below
![230539](https://user-images.githubusercontent.com/25908688/97205437-2b25f680-17c0-11eb-9b38-b7a996cf7e9a.png)
I think this mainly because of CARLA server itself, what do you think?

2) The second problem is missing Bboxes due to the area that you have chosen becomes out of field of view.
I think we could easily solve it by using the segmentation mask instead of generating smaller box from the original one.
![231251](https://user-images.githubusercontent.com/25908688/97205578-53adf080-17c0-11eb-8179-7165f778076e.png)

![231175](https://user-images.githubusercontent.com/25908688/97205567-514b9680-17c0-11eb-95ab-c2e43691396c.png)

**Comment by MukhlasAdib** [comment]
Hi again,

1. I also think that it is caused by unsolved error in CARLA synchronization. I have encountered the same problem quite often when collecting multiple sensors' data in CARLA and it seems that it has become a long issue in CARLA community itself. However, my module is not responsible for the data generation process. So, it might require some tricky coding but maybe it is something that can be solved in client side. (Note: I haven't solved it either)

2. Thanks for informing me about this. I used the vehicle's centre coordinate to determine whether it is inside the camera FOV or not. So, even though that most of vehicle's body is visible, it will be filtered out if its centre point is not inside the FOV. And also thanks for your suggestion. I'll fix it as soon as I can.

Anyway, I have just read CARLA 0.9.10 update info and I found that this version provides LIDAR instance segmentation. I think it will be more efficient to use it for BBox extraction rather than using depth image. It might also solve the problem No. 2 that you mentioned. I'm working on it.

**Comment by eslambakr** [comment]
Hello @MukhlasAdib 

I have solved the first problem as we said it was because of synchronization problem.
I noticed that you began your code "test_draw_bb.py" from CARLA example code "client_bounding_boxes.py" which is suffering from synchronization problem, so I re-base the code to start from "synchronous_mode.py" instead, which illuminate the shifted boxes problem.

I am working now on solving the missing boxes problem but using the same version 9.8 to avoid the hassle of upgrading CARLA version.
I will keep the issue open and post the updates here, as it may be useful for others
